File: LeetCode/28.ImplementstrStr().cpp
// https://leetcode.com/problems/implement-strstr/
// Brute force 
class Solution {
public:
    int strStr(string haystack, string needle) {
        
        int lens = haystack.size(), lent = needle.size();
        if(lent == 0 ) return 0;
        for(int i = 0; i <= lens - lent; ++i){
            for(int j = 0; j < lent;++j){
                if(haystack[i+j] != needle[j]){
                    break;
                }
                if(lent - 1 == j){
                    return i;
                }
            }
        }
        return -1;
    }
};

File: LeetCode/Python3/28.ImplementstrStr().py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        l1 = len(haystack)
        l2 = len(needle)
        for i in range(l1-l2+1):
            if haystack[i:i+l2] == needle:
                return i
        return -1
